lib/v2/zhihu/question.js

```diff
@@ -1,22 +1,36 @@
 const got = require('@/utils/got');
 const utils = require('./utils');
-const config = require('@/config').value;
 const md5 = require('@/utils/md5');
 const g_encrypt = require('./execlib/g_encrypt');
 const { parseDate } = require('@/utils/parse-date');
 
 module.exports = async (ctx) => {
-    const cookie = config.zhihu.cookies_no_login;
-    if (!cookie) {
-        throw Error('缺少未登录状态下的知乎 Cookie 值(dc_0) (请在环境变量中设置ZHIHU_COOKIES_NO_LOGIN)'); // the key is "d_c0"
-    }
+    const { questionId } = ctx.params;
+    // Because the API of zhihu.com changed,we must use the dc_0(get from cookie without logging) to calculate the x-zse-96. So we first to get the dc_0,then we get the real-data of ZhiHu question. In this way, we don't need set the cookie in .env anymore
+    // fisrt: get cookie(dc_0) from zhihu.com
+    const response1 = await got({
+        method: 'get',
+        url: `https://www.zhihu.com/question/${questionId}`,
+        headers: {
+            ...utils.header,
+            Referer: `https://www.zhihu.com/question/${questionId}`,
+        },
+    });
+
+    /*
+     *Fixme:
+     *I'm not familiar with RSSHub, but I think the cookie should store in someplace(like rsshub cache?) to avoid get cookie from zhihu.com everytime.
+     *maybe someone can improve this.
+     */
+    const cookie_org = response1.headers['set-cookie'];
+    const cookie = cookie_org.toString();
     const match = cookie.match(/d_c0=(\S+?)(?:;|$)/);
     const cookie_mes = match && match[1];
     if (!cookie_mes) {
-        throw Error('请确认环境变量ZHIHU_COOKIES_NO_LOGIN包含"dc_0"字段)'); // the key is "d_c0"
+        throw Error('未能获取cookie"dc_0"字段');
     }
```

Comment:
```suggestion
    const cookie_mes = await ctx.cache.tryGet('zhihu:cookies:dc_0', async () => {
        const response = await got({
            method: 'get',
            url: `https://www.zhihu.com/question/${questionId}`,
            headers: {
                ...utils.header,
            },
        });

        const cookie_org = response.headers['set-cookie'];
        const cookie = cookie_org.toString();
        const match = cookie.match(/d_c0=(\S+?)(?:;|$)/);
        const cookie_mes = match && match[1];
        if (!cookie_mes) {
            throw Error('Failed to extract `dc_0` from cookies');
        }
        return cookie_mes;
    });
    const cookie = `d_c0=${cookie_mes}`;
```

Reply:
ah，that is! thanks a lot!

Reply:
Applying suggestions on deleted lines is currently not supported.

Reply:
> Applying suggestions on deleted lines is currently not supported.

You may commit it locally.

Reply:
it's ok now.